hudi-utilities/src/main/java/org/apache/hudi/utilities/streamer/StreamSync.java

```diff
@@ -456,6 +457,7 @@ HoodieTableMetaClient initializeEmptyTable(HoodieTableMetaClient.TableBuilder ta
             Boolean.parseBoolean(HIVE_STYLE_PARTITIONING_ENABLE.defaultValue())))
         .setUrlEncodePartitioning(props.getBoolean(URL_ENCODE_PARTITIONING.key(),
             Boolean.parseBoolean(URL_ENCODE_PARTITIONING.defaultValue())))
+        .setTableFormat(props.getProperty(HoodieTableConfig.TABLE_FORMAT.key(), NativeTableFormat.TABLE_FORMAT))
```

Comment:
ensure every writer path - has this property wired in. i.e wherever the table version is set.